libraries/standard/mqtt/src/mqtt_state.c

```diff
@@ -284,31 +486,48 @@ static MQTTStatus_t addRecord( MQTTPubAckInfo_t * records,
     MQTTStatus_t status = MQTTNoMemory;
     int32_t index = 0;
     size_t availableIndex = recordCount;
+    bool validEntryFound = false;
 
     assert( packetId != MQTT_PACKET_ID_INVALID );
     assert( qos != MQTTQoS0 );
 
-    /* Start from end so first available index will be populated. */
+    /* Check if we have to compact the records. This is known by checking if
+     * the last spot in the array is filled. */
+    if( records[ ( recordCount - 1U ) ].packetId != MQTT_PACKET_ID_INVALID )
```

Comment:
Don't really need parentheses
```suggestion
    if( records[ recordCount - 1U ].packetId != MQTT_PACKET_ID_INVALID )
```